packages/runtime/src/next/components/head-tags.tsx

```diff
@@ -0,0 +1,66 @@
+import { ComponentPropsWithoutRef } from 'react'
+import { useIsPagesRouter } from '../../components/hooks/useIsPagesRouter'
+import Head from 'next/head'
+
+export function PageTitle({ children, ...props }: ComponentPropsWithoutRef<'title'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <title {...props}>{children}</title>
+      </Head>
+    )
+  }
+  return <title {...props}>{children}</title>
```

Comment:
Note that for `<title>` in React Canary, `children` must be a string. This means that `ComponentPropsWithoutRef<'title'>` might not be the appropriate type.